controllers/CartController.js

```diff
@@ -0,0 +1,201 @@
+                
+                if (createCart) res.status(201).json({ message: 'Cart successfully added' })
+            } else {
+                const findProduct = await Product.findOne({
+                    where: {
+                        id: payload.ProductId
+                    }
+                })
+
+                const { qty } = findMyCart
+                const { stock } = findProduct
+
+                if (qty + 1 > stock) {
+                    throw new Error('Exceeds the available stock')
+                } else {
+                    findMyCart.qty ++
+                    await findMyCart.save()
+                    res.status(200).json({ message: 'Successfuly Added Cart Quantity'})
+                }
+            }
+        } catch (error) {
+            next(error)
+        }
+    }
+
+    static async patchMyCart (req, res, next) {
+        try {
+            const { increment } = req.body
+            const id = { id: req.params.id }
+            console.log(increment, id, 'di cartttt')
+
+            const findMyCart = await UserCart.findOne({
+                where: id
+            })
+
+            const ProductId = { id: findMyCart.ProductId }
+
+            const findProduct = await Product.findOne({
+                where: ProductId
+            })
+
+            const { stock } = findProduct
+            const { qty } = findMyCart
+
+            if (increment) {
+                if (qty + 1 > stock) throw new Error('Exceeds the available stock')
+                findMyCart.qty ++
+                await findMyCart.save()
+                res.status(200).json({ message: 'Successfuly Updated Cart Quantity'})
+            } else {
+                if (qty - 1 <= 0) throw new Error('Less than available stock')
+                findMyCart.qty --
+                await findMyCart.save()
+                res.status(200).json({ message: 'Successfuly Updated Cart Quantity'})
+            }
+        } catch (error) {
+            next(error)
+        }
+    }
+
+    static async deleteMyCart (req, res, next) {
+        try {
+            const id = { id: req.params.id }
+            await UserCart.destroy({
+                where: id
+            })
+            
+            res.status(200).json({ message: 'Successfuly Deleted Cart' })
+        } catch (error) {
+            next(error)
+        }
+    }
+
+    static async checkoutCart (req, res, next) {
+        try {
+            const { carts } = req.body
+            const UserId = req.whoAmI.id
+            const { email } = req.whoAmI
+            const uuid = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyx0123456789'
+            const items = []
+            let invoiceId = ''
+
+            for (let i = 0; i < 5; i++) {
+                invoiceId += uuid[Math.floor(Math.random() * uuid.length)]
+            }
+
+            // * START OF LOOPING CART
+            await sequelize.transaction(async t => {
```

Comment:
mantap dah pake transaction

Reply:
Ini juga untuk memahami sistem Flash Sale kak, akhirnya saya tau bagaimana flow apabila banyak user ingin membeli produk yang sama secara bersamaan. Thank you ka! ^^